### PR TITLE
allow commands to be attached to watchpoints

### PR DESCRIPTION
**Pull request by joooooooooooooooooooooooooooooooooooosh**

### CODE REVIEW

File: crates/mipsy_interactive/src/interactive/commands/mod.rs [hunk 1]
```diff
@@ -21,7 +21,6 @@ mod watchpoint;
 pub(crate) use back::back_command;
 pub(crate) use breakpoint::breakpoint_command;
 pub(crate) use context::context_command;
-pub(crate) use commands::commands_command;
```
Comment (Dylan-Brotherston) [inline crates/mipsy_interactive/src/interactive/commands/mod.rs:24]: what a concise and not at all tautological name.

File: crates/mipsy_interactive/src/interactive/commands/watchpoint.rs [hunk 2]
```diff
@@ -66,6 +70,8 @@ fn get_long_help() -> String {
         "A collection of commands for managing watchpoints. Available {10}s are:\n\n\
          {0} {2}    : insert/delete a watchpoint\n\
          {1} {3}\n\
+         {0} {12} : insert a temporary breakpoint that deletes itself after being hit\n\
+         {0} {13}  : attach commands to a breakpoint\n\
```
Comment (Dylan-Brotherston) [inline crates/mipsy_interactive/src/interactive/commands/watchpoint.rs:74]: watchpoint

File: crates/mipsy_interactive/src/interactive/commands/watchpoint.rs [hunk 3]
```diff
@@ -66,6 +70,8 @@ fn get_long_help() -> String {
         "A collection of commands for managing watchpoints. Available {10}s are:\n\n\
          {0} {2}    : insert/delete a watchpoint\n\
          {1} {3}\n\
+         {0} {12} : insert a temporary breakpoint that deletes itself after being hit\n\
```
Comment (Dylan-Brotherston) [inline crates/mipsy_interactive/src/interactive/commands/watchpoint.rs:73]: watchpoint

File: crates/mipsy_lib/src/compile/mod.rs [hunk 4]
```diff
@@ -60,6 +60,21 @@ impl Breakpoint {
     }
 }
 
+pub trait Point {
+    fn get_id(&self) -> u32;
+    fn get_commands(&'_ mut self) -> &'_ mut Vec<String>;
+}
+
+impl Point for Breakpoint {
```
Comment (Dylan-Brotherston) [inline crates/mipsy_lib/src/compile/mod.rs:68]: Why is the `impl Point for Breakpoint` in `crates/mipsy_lib/src/compile/mod.rs` and `impl Point for Watchpoint` in `crates/mipsy_interactive/src/interactive/commands/mod.rs`?
They seem like they should be in similar locations.

File: crates/mipsy_interactive/src/interactive/commands/breakpoint.rs [hunk 5]
```diff
@@ -1,7 +1,7 @@
 use crate::interactive::{error::CommandError, prompt};
 use std::iter::successors;
 
-use super::*;
+use super::{*, commands::handle_commands};
```
Comment (Dylan-Brotherston) [inline crates/mipsy_interactive/src/interactive/commands/breakpoint.rs:4]: is `commands::handle_commands` needed if there is already a glob import?

File: crates/mipsy_interactive/src/interactive/commands/watchpoint.rs [hunk 6]
```diff
@@ -1,7 +1,7 @@
 use crate::interactive::{error::CommandError, prompt};
 use std::{iter::successors, str::FromStr};
 
-use super::*;
+use super::{*, commands::handle_commands};
```
Comment (Dylan-Brotherston) [inline crates/mipsy_interactive/src/interactive/commands/watchpoint.rs:4]: is `commands::handle_commands` needed if there is already a glob import?

Comment (joooooooooooooooooooooooooooooooooooosh) [reply crates/mipsy_interactive/src/interactive/commands/watchpoint.rs:4]: you would think not but apparently it is. my guess is the glob import doesn't import recursively

File: crates/mipsy_interactive/src/interactive/commands/watchpoint.rs (hunk 2 above)
Comment (joooooooooooooooooooooooooooooooooooosh) [reply crates/mipsy_interactive/src/interactive/commands/watchpoint.rs:74]: whoops

File: crates/mipsy_interactive/src/interactive/commands/watchpoint.rs (hunk 6 above)
Comment (Dylan-Brotherston) [reply crates/mipsy_interactive/src/interactive/commands/watchpoint.rs:4]: rust needs `use super::**;`

File: crates/mipsy_lib/src/compile/mod.rs (hunk 4 above)
Comment (Dylan-Brotherston) [reply crates/mipsy_lib/src/compile/mod.rs:68]: mipsy_web may want to implement watchpoints.
So having `impl Point for Watchpoint` in lib also might be a good idea.

Comment (joooooooooooooooooooooooooooooooooooosh) [reply crates/mipsy_lib/src/compile/mod.rs:68]: ah I hadn't considered this, I had `Watchpoint` in `mipsy_interactive` as I didn't think `mipsy_lib` would have a use for them, but this is definitely worth adding to `mipsy_web`

**Review by Dylan-Brotherston — APPROVED** [review]

**Pull request merged**